toolkit/scripts/setuplkgtoolchain.sh

```diff
@@ -23,39 +28,97 @@ get_lkg() {
+        fi
     fi
 }
 
-update_manifests() {
-    wget -nv https://raw.githubusercontent.com/microsoft/CBL-Mariner/$GIT_COMMIT/toolkit/resources/manifests/package/toolchain_$(uname -p).txt -O $ROOT_FOLDER/toolkit/resources/manifests/package/toolchain_$(uname -p).txt
-    wget -nv https://raw.githubusercontent.com/microsoft/CBL-Mariner/$GIT_COMMIT/toolkit/resources/manifests/package/pkggen_core_$(uname -p).txt -O $ROOT_FOLDER/toolkit/resources/manifests/package/pkggen_core_$(uname -p).txt
+# We don't want to update the timestamps unless we actually change the file, so download to a temp file and then move it over
+# if the files are different
+# $1 - filename (e.g. toolchain_aarch64.txt)
+update_manifest() {
+    local filename
+    local url
+    local filepath
+    filename="$1"
+    url="https://raw.githubusercontent.com/microsoft/CBL-Mariner/$GIT_COMMIT/toolkit/resources/manifests/package/${filename}"
+    filepath="$ROOT_FOLDER/toolkit/resources/manifests/package/${filename}"
+    filepath_tmp="${filepath}.tmp"
+
+    echo "Downloading $filename manifest from $url" 1>&2
+
+    wget -nv "${url}" -O "${filepath_tmp}" 1>&2
+    # only update if the files are different
+    if ! cmp -s "${filepath_tmp}" "${filepath}"; then
+        echo "Manifests are different, updating ${filename}" 1>&2
+        mv "${filepath_tmp}" "${filepath}"
+        chown "$CALLING_USER:$CALLING_USER" "${filepath}"
+    else
+        rm -f "${filepath_tmp}"
+    fi
 }
 
 cleanup() {
     rm -f $LKG_FILENAME
 }
 
-[[ "$1" == "--help" || "$1" == "help" ]] && usage
+auto_id=false
+force_manifests=false
+skip_manifest_updates=false
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        *auto-id)
+            auto_id=true
+            shift
+            ;;
+        *force)
+            force_manifests=true
+            shift
+            ;;
+        *skip-manifest-updates)
+            skip_manifest_updates=true
+            shift
+            ;;
+        *help | *)
+            usage
+            ;;
+    esac
+done
 
 get_lkg
 
-check_for_modified_manifests
-
-update_manifests
+if ! $skip_manifest_updates; then
+    check_for_modified_manifests $auto_id $force_manifests
+    update_manifest "toolchain_$(uname -p).txt"
+    update_manifest "pkggen_core_$(uname -p).txt"
+fi
 
+if ! $auto_id; then
 cat << EOF
 ===
 === Current LKG:
 === DAILY_BUILD_ID='$DAILY_BUILD_ID'
 === Commit='$GIT_COMMIT'
 ===
 EOF
-
-echo -e "Finished syncing toolchain to LKG ('$DAILY_BUILD_ID' - '$GIT_COMMIT')"
-echo -e "To download LKG toolchain, run:\n\tsudo make toolchain -j$(nproc) REBUILD_TOOLCHAIN=n REBUILD_TOOLS=y DAILY_BUILD_ID=$DAILY_BUILD_ID"
+    echo -e "Finished syncing toolchain to LKG ('$DAILY_BUILD_ID' - '$GIT_COMMIT')"
+    echo -e "To download LKG toolchain, run:\n\tsudo make toolchain -j$(nproc) REBUILD_TOOLCHAIN=n REBUILD_TOOLS=y DAILY_BUILD_ID=$DAILY_BUILD_ID"
+else
+    # In auto mode, just print the LKG build ID so we can use it in the calling script
+    echo $DAILY_BUILD_ID
+fi
```

Comment:
I'm a worried about this bit, that it's now tightly coupling a script that can be run independently with one specific use in our toolkit. Would it be problematic to have the script print its original output and add a grep in the caller like `grep -oP (?<=DAILY_BUILD_ID=')[^']+`?

Reply:
I thought about it a bit more and I think the fact we're not making this tool return two different things required for the toolkit to function (update the manifests and return the build ID, which is now an input for something else) maybe causing us headaches now (echoing two different output depending on `auto_id` and forcing us to print to `stderr`) and in the future.

In addition to that, we're not introducing a new way for the toolkit to update/create files that is I think also unusual for `make` builds. Instead of using a target, we have a shell call done while we assign a value to the `DAILY_BUILD_ID` argument. This makes it even harder to understand what's happening in the `Makefile`.

What do you think about splitting the logic:
- Have a separate target, which updates the manifests (and prints whatever information or warnings it likes). This way we also won't have to check its error code, since `make` will do it for us.
- Have a separate script dedicated only to returning the build ID or other values but making no file changes. This is still not ideal (the script may still fail and require a check of the error code) but I think it's a consequence of using URLs directly encoded inside arguments like `PACKAGE_URL_LIST`. Maybe we can remedy that somehow later (move the URLs to files?).

Reply:
I've split these off, and added a dedicated `dailly_build.mk` file.